execution/executor/src/block_executor/block_tree/mod.rs

```diff
@@ -205,6 +209,21 @@ impl BlockTree {
         self.block_lookup.multi_get(ids)
     }
 
+    fn is_new_virtual_genesis(db: &Arc<dyn DbReader>) -> Result<bool> {
+        let db_state_view = db
+            .latest_state_checkpoint_view()
+            .map_err(|err| anyhow!("[aptos-node] failed to create db state view {}", err))?;
+        let chain_id = ChainIdResource::fetch_config(&db_state_view)
+            .expect("[aptos-node] missing chain ID resource")
+            .chain_id();
+        let ledger_info_with_sigs = db.get_latest_ledger_info()?;
+        let cur_epoch = ledger_info_with_sigs.ledger_info().epoch();
+        if chain_id.is_movement_mainnet() && cur_epoch == 1848695 {
```

Comment:
The `1848695` should be explained, and it's not compatible with a testnet. The L1 testnet will have the same chain_id or we are going to start from a fresh new DB?

Reply:
update the logic to passing the new block id from consensus to execution. so this magic number is no longer needed.